rir/src/compiler/opt/force_dominance.cpp

```diff
@@ -89,8 +94,17 @@ class ForceDominanceAnalysis : public StaticAnalysis<ForcedAt> {
 
     void apply(ForcedAt& d, Instruction* i) const override {
         auto f = Force::Cast(i);
-        if (f)
-            d.evalAt(getValue(f), f);
+        if (f) {
+            MkArg* arg = MkArg::Cast(getValue(f));
+            if (arg)
```

Comment:
`if (arg)` on line 99

Reply:
but I don't understand, why wouldn't it be necessary? We want to figure out where known promises are forced. and we only know if we see the mkarg.

Reply:
Well it is probably just my misunderstanding, but isn't it always true that value of a force instruction is a mkarg instruction? That is the only thing I meant. The if(arg) is redundant because it will always be true with my assumption

Reply:
ah, no. you can `force` anything that you are not sure is a value. for example `LdArg` returns a value which may or may not be evaluated. The type is `value^`, the `^` indicates it might be lazy, the `force` removes the `^`, so to speak.